src/DocumentDB.ChangeFeedProcessor/LeaseManagement/DocumentServiceLeaseStoreManager.cs

```diff
@@ -107,9 +109,11 @@ public async Task<ILease> CreateLeaseIfNotExistAsync(string partitionId, string
                 ContinuationToken = continuationToken,
             };
 
+            var leaseDoc = DocumentServiceLease.StampWithCustomPartitionKeyProperty(documentServiceLease, this.settings.LeaseCollectionPartitionKeyPropertyName);
+
             bool created = await this.client.TryCreateDocumentAsync(
                 this.settings.LeaseCollectionLink,
-                documentServiceLease).ConfigureAwait(false) != null;
+                leaseDoc).ConfigureAwait(false) != null;
             if (created)
             {
                 Logger.InfoFormat("Created lease for partition {0}.", partitionId);
```

Comment:
returning documentServiceLease seems to be wrong. Better would be on L112 assign the result to documentServiceLease and not introduce the leaseDoc

Reply:
when we stamp, it becomes JObject and cant be DocumentServiceLease object. we cant return JObject for ILease return type.